### PR TITLE
Add ``WeightedIngredient`` etc.

### PR DESCRIPTION
**Pull request by GBLodb**
Title, the PR introduced/reworked some weighted related stuff.

### CODE REVIEW

File: CraftTweaker2-API/src/main/java/crafttweaker/api/liquid/WeightedLiquidStack.java [hunk 1]
```diff
@@ -11,9 +12,9 @@
  *
  * @author Stan Hebben
  */
-@ZenClass("crafttweaker.item.WeightedLiquidStack")
+@ZenClass("crafttweaker.liquid.WeightedLiquidStack")
```
Comment (jaredlll08) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/liquid/WeightedLiquidStack.java]: While I understand this is a typo, we can't change it now without potentially breaking a bunch of scripts, so this needs to be reverted.

**Review by jaredlll08 — CHANGES_REQUESTED** [review]

File: CraftTweaker2-API/src/main/java/crafttweaker/api/liquid/WeightedLiquidStack.java (hunk 1 above)
Comment (GBLodb) [reply CraftTweaker2-API/src/main/java/crafttweaker/api/liquid/WeightedLiquidStack.java]: > While I understand this is a typo, we can't change it now without potentially breaking a bunch of scripts, so this needs to be reverted.

@jaredlll08 Done, changed the class name back.

**Review by jaredlll08 — APPROVED** [review]

**Comment by jaredlll08** [comment]
Thanks!

**Comment by jaredlll08** [comment]
Just a heads up for the future, please load the game with your changes.
https://github.com/CraftTweaker/CraftTweaker/pull/1529/files#diff-9928590284e0f7d73b4e96a7bc5acc4ee5a56faf6a7327b3fa53314b8028c0cfR14
should have been:
https://github.com/CraftTweaker/CraftTweaker/commit/548c708ca6e9ae7e08ab926f435b65aa2feb2e5b

**Pull request merged**